src/utils/account.ts

```diff
@@ -35,123 +40,183 @@ export const getAccount = async (args: {
-
-      if (!gcpApplicationCredentialEmail) {
-        throw new Error(
-          "GCP application credential email is missing for the wallet",
-        );
-      }
-
-      return await getGcpKmsAccount({
+      const account = await getGcpKmsAccount({
         client: thirdwebClient,
-        name: resourcePath,
+        name: walletDetails.gcpKmsResourcePath,
         clientOptions: {
           credentials: {
-            client_email: gcpApplicationCredentialEmail,
-            private_key: gcpApplicationCredentialPrivateKey,
+            client_email: walletDetails.gcpApplicationCredentialEmail,
+            private_key: walletDetails.gcpApplicationCredentialPrivateKey,
           },
         },
       });
+      return { account };
     }
     case WalletType.local: {
-      if (accountAddress) {
-        const chain = await getChain(chainId);
-        const account = await getSmartWalletV5({
-          chain,
-          accountAddress,
-          from,
-        });
-        _accountsCache.set(cacheKey, account);
-        return account;
-      }
-
-      const account = await getLocalWalletAccount(from);
-      _accountsCache.set(cacheKey, account);
-      return account;
+      const account = await getLocalWalletAccount(
+        getAddress(walletDetails.address),
+      );
+      return { account };
+    }
+    case WalletType.smartAwsKms: {
+      const { keyId, region } = splitAwsKmsArn(walletDetails.awsKmsArn);
+
+      const signerAccount = await getAwsKmsAccount({
+        client: thirdwebClient,
+        keyId,
+        config: {
+          region,
+          credentials: {
+            accessKeyId: walletDetails.awsKmsAccessKeyId,
+            secretAccessKey: walletDetails.awsKmsSecretAccessKey,
+          },
+        },
+      });
+
+      const unconnectedSmartWallet = smartWallet({
+        chain,
+        sponsorGas: true,
+        factoryAddress: walletDetails.accountFactoryAddress ?? undefined,
+      });
+
+      const account = await unconnectedSmartWallet.connect({
+        client: thirdwebClient,
+        personalAccount: signerAccount,
+      });
+
+      return { account, adminAccount: signerAccount };
+    }
+
+    case WalletType.smartGcpKms: {
+      const signerAccount = await getGcpKmsAccount({
+        client: thirdwebClient,
+        name: walletDetails.gcpKmsResourcePath,
+        clientOptions: {
+          credentials: {
+            client_email: walletDetails.gcpApplicationCredentialEmail,
+            private_key: walletDetails.gcpApplicationCredentialPrivateKey,
+          },
+        },
+      });
+
+      const unconnectedSmartWallet = smartWallet({
+        chain,
+        sponsorGas: true,
+        factoryAddress: walletDetails.accountFactoryAddress ?? undefined,
+      });
+
+      const account = await unconnectedSmartWallet.connect({
+        client: thirdwebClient,
+        personalAccount: signerAccount,
+      });
+
+      return { account, adminAccount: signerAccount };
```

Comment:
Nit: signerAccount and adminAccount should mean the same thing, so keep consistent and use one name everywhere. (No strong opinion which one)

I really want to avoid multiple names that mean the same thing since it's easy to get tripped up (we already have so many addresses to keep track of).